bundles/org.eclipse.ui.ide/src/org/eclipse/ui/internal/wizards/datatransfer/WizardFileSystemResourceImportPage1.java

```diff
@@ -1122,9 +1123,7 @@ public void filterElements(Collection files,
 				}
 				Iterator filesList = files.iterator();
 				while (filesList.hasNext()) {
-					if (monitor.isCanceled()) {
-						throw new InterruptedException();
-					}
+					ModalContext.checkCanceled(monitor);
```

Comment:
This is only okay when not changing the `ModalContext.checkCanceled` implementation, as the exception is only used with this method (line 1170), so no need to add overhead.